databuilder/publisher/neo4j_csv_publisher.py

```diff
@@ -273,15 +293,33 @@ def _publish_relation(self, relation_file):
         :return:
         """
 
-        tx = self._session.begin_transaction()
+        if self._relation_preprocessor.is_perform_preprocess():
+            LOGGER.info('Pre-processing relation with {}'.format(self._relation_preprocessor))
+
+            count = 0
+            with open(relation_file, 'r') as relation_csv:
+                for rel_record in csv.DictReader(relation_csv):
+                    stmt, params = self._relation_preprocessor.preprocess_cypher(
```

Comment:
does it expect to return a tuple( stmt, params ) for all the call?

Reply:
Yes.
Cypher query improves a lot using query parameter. I am introducing that pattern from this PR.